Zero Rust warnings

ckauhaus opened the pull request:
Go down to zero warnings for Rust 1.43 and adjust formatting to comply with current rustfmt default settings. This change does not have much value in its own. But having more idiomatic Rust code makes it easier for others to contribute and minizies diffs in the future.

Minimum supported Rust version is now 1.36 due to the use of std::mem::MaybeUninit.

Some of the things done:

- Fix redundant struct initializers
- Fix redundant imports
- Fix unnecessary to_owned() calls
- Fix single-char str patterns
- Use ok_or_else() to avoid unnecessary code execution on the unhappy
  path
- Fix repo url in Cargo.toml

Also reformat the whole source with `cargo fmt` in its default settings.

ckauhaus commented:
Releasing a new version of probes-rs would probably a good idea given the recent improvements accepted into master.

ckauhaus commented:
I think this PR is still relevant, but someone from the project team needs to review it.

tombruijn commented:
@thijsc or @matsimitsu can you review? You know more about Rust :)

thijsc commented:
Thanks for your work on this!

Curious: What are you using this library for?

tombruijn commented:
@thijsc should the changes since the last release be released as a patch or minor release? Diff: https://github.com/appsignal/probes-rs/compare/v0.3.0...main

ckauhaus commented:
@thijsc I use probes::load for load dependent throttling in a filesystem scanning application

ckauhaus commented:
> @thijsc should the changes since the last release be released as a patch or minor release? Diff: [v0.3.0...main](https://github.com/appsignal/probes-rs/compare/v0.3.0...main)

I'd suggest 0.3.1
